flink-end-to-end-tests/test-scripts/common_mesos_docker.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+################################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+set -o pipefail
+
+source "$(dirname "$0")"/common.sh
+
+MAX_RETRY_SECONDS=120
+IMAGE_BUILD_RETRIES=5
+
+export INPUT_VOLUME=${END_TO_END_DIR}/test-scripts/test-data
+
+echo "End-to-end directory $END_TO_END_DIR"
+docker --version
+docker-compose --version
+
+start_time=$(date +%s)
+
+# make sure we stop our cluster at the end
+function cluster_shutdown {
+  docker-compose -f $END_TO_END_DIR/test-scripts/docker-mesos-cluster/docker-compose.yml down
+  sudo rm -rf ${INPUT_VOLUME}/log
+  sudo rm -rf ${INPUT_VOLUME}/tmp
+}
+on_exit cluster_shutdown
+
+function wait_flink_running_on_mesos {
+  # wait at most 10 seconds until the dispatcher is up
+  local QUERY_URL="http://mesos-master:8081/taskmanagers"
+  local TIMEOUT=20
+  for i in $(seq 1 ${TIMEOUT}); do
+    # without the || true this would exit our script if the JobManager is not yet up
+    QUERY_RESULT=$(docker exec -it mesos-master curl "$QUERY_URL" 2> /dev/null || true)
+
+    # ensure the taskmanagers field is there at all and is not empty
+    if [[ ${QUERY_RESULT} =~ \{\"taskmanagers\":\[.*\]\} ]]; then
+      echo "Dispatcher REST endpoint is up."
+      return
+    fi
+
+    echo "Waiting for dispatcher REST endpoint to come up..."
+    sleep 1
+  done
+  echo "Dispatcher REST endpoint has not started within a timeout of ${TIMEOUT} sec"
+  exit 1
+}
+
+function start_flink_cluster_with_mesos() {
+    echo "Starting Flink on Mesos cluster"
+    build_image
+
+    docker-compose -f $END_TO_END_DIR/test-scripts/docker-mesos-cluster/docker-compose.yml up -d
+
+    # wait for the Mesos master and slave set up
+    start_time=$(date +%s)
+    until docker logs mesos-master 2>&1 | grep -q "Successfully recovered registrar"
```

Comment:
Yes, we can ;). I think we still need to wait for the master and the slave. At least it could help to clarify which component going wrong when tests fail.